File: aspnet-core/src/ProductManager.Application/Products/Dto/ProductDto.cs
﻿using System.ComponentModel.DataAnnotations;
using Abp.Application.Services.Dto;
using Abp.AutoMapper;
using ProductManager.Entities;

namespace ProductManager.Products.Dto
{
    [AutoMap(typeof(Product))]
    public class ProductDto : FullAuditedEntityDto
    {
        [Required] public string Code { get; set; }
        [Required] public string Name { get; set; }
        public string Photo { get; set; }

        [Required]
        [RegularExpression(@"^(((\d{1,3})(,\d{3})*)|(\d+))(.\d+)?$")]
        [Range(0, int.MaxValue)]
        public decimal Price { get; set; }
    }
}

File: reactjs/src/services/product/dto/productDto.ts
import { EntityDto } from '../../../services/dto/entityDto';

export default class ProductDto extends EntityDto {
  code!: string;
  name!: string;
  photo!: string;
  price!: number;
  isDeleted!: boolean;
  deleterUserId!: number;
  deletionTime!: Date;
  lastModificationTime!: Date;
  lastModifierUserId!: number;
  creationTime!: Date;
  creatorUserId!: number;
}
